lib/internal/errors.js

```diff
@@ -1113,16 +1112,9 @@ E('ERR_INVALID_FILE_URL_PATH', 'File URL path %s', TypeError);
 E('ERR_INVALID_HANDLE_TYPE', 'This handle type cannot be sent', TypeError);
 E('ERR_INVALID_HTTP_TOKEN', '%s must be a valid HTTP token ["%s"]', TypeError);
 E('ERR_INVALID_IP_ADDRESS', 'Invalid IP address: %s', TypeError);
-E('ERR_INVALID_MODULE_SPECIFIER', (pkgPath, subpath, base = undefined) => {
-  if (subpath === undefined) {
-    return `Invalid package name '${pkgPath}' imported from ${base}`;
-  } else if (base === undefined) {
-    assert(subpath !== '.');
-    return `Package subpath '${subpath}' is not a valid module request for ` +
-      `the "exports" resolution of ${pkgPath}${sep}package.json`;
-  }
-  return `Package subpath '${subpath}' is not a valid module request for ` +
-      `the "exports" resolution of ${pkgPath} imported from ${base}`;
+E('ERR_INVALID_MODULE_SPECIFIER', (request, reason, base = undefined) => {
+  return `Invalid module "${request}" ${reason}${base ?
```

Comment:
This looks to regress the original error, dropping the information on which `package.json` to check. Can we preserve this?

Reply:
This retains the original information in the `reason` spot so the message is the same for all previous messages (see eg https://github.com/nodejs/node/pull/34117/files#diff-0c6f26a01443323e777b174ae24c4bf5R297), but the reason this was changed is because I fixed up the case for `import('./%2F')` to directly give the invalid module specifier error for having encoded `/` characters instead of the slightly more cryptic file URL error that doesn't include context. This error doesn't have an associated package.json. In theory this was a spec bug fix.